### ISSUE TITLE
0.9.0: error: reference to type 'const std::shared_ptr<const void>' could not bind to an rvalue of type 'const SymEngine::Basic *'

### ISSUE THREAD

**Issue by yurivict** [symengine/symengine#1907]
clang-13 prints this error:
```
In file included from /disk-samsung/freebsd-ports/math/symengine/work/symengine-0.9.0/symengine/basic.cpp:3:
/disk-samsung/freebsd-ports/math/symengine/work/symengine-0.9.0/symengine/serialize-cereal.h:288:44: error: reference to type 'const std::shared_ptr<const void>' could not bind to an rvalue of type 'const SymEngine::Basic *'
    uint32_t id = ar.registerSharedPointer(ptr.get());
                                           ^~~~~~~~~
/disk-samsung/freebsd-ports/math/symengine/work/symengine-0.9.0/symengine/serialize-cereal.h:311:5: note: in instantiation of function template specialization 'SymEngine::save_basic<cereal::PortableBinaryOutputArchive>' requested here
    save_basic(ar, rcp_static_cast<const Basic>(ptr));
    ^
/usr/local/include/cereal/cereal.hpp:532:9: note: in instantiation of function template specialization 'SymEngine::save<cereal::PortableBinaryOutputArchive, SymEngine::Basic>' requested here
        CEREAL_SAVE_FUNCTION_NAME(*self, t);
        ^
/usr/local/include/cereal/macros.hpp:92:35: note: expanded from macro 'CEREAL_SAVE_FUNCTION_NAME'
#define CEREAL_SAVE_FUNCTION_NAME save
                                  ^
/usr/local/include/cereal/cereal.hpp:444:15: note: in instantiation of function template specialization 'cereal::OutputArchive<cereal::PortableBinaryOutputArchive, 1>::processImpl<SymEngine::RCP<const SymEngine::Basic>, 0>' requested here
        self->processImpl( head );
              ^
/usr/local/include/cereal/cereal.hpp:453:15: note: in instantiation of function template specialization 'cereal::OutputArchive<cereal::PortableBinaryOutputArchive, 1>::process<SymEngine::RCP<const SymEngine::Basic>>' requested here
        self->process( std::forward<Other>( tail )... );
              ^
/usr/local/include/cereal/cereal.hpp:453:15: note: in instantiation of function template specialization 'cereal::OutputArchive<cereal::PortableBinaryOutputArchive, 1>::process<unsigned short &, SymEngine::RCP<const SymEngine::Basic>>' requested here
/usr/local/include/cereal/cereal.hpp:333:15: note: in instantiation of function template specialization 'cereal::OutputArchive<cereal::PortableBinaryOutputArchive, 1>::process<unsigned short &, unsigned short &, SymEngine::RCP<const SymEngine::Basic>>' requested here
        self->process( std::forward<Types>( args )... );
              ^
/disk-samsung/freebsd-ports/math/symengine/work/symengine-0.9.0/symengine/basic.cpp:56:45: note: in instantiation of function template specialization 'cereal::OutputArchive<cereal::PortableBinaryOutputArchive, 1>::operator()<unsigned short &, unsigned short &, SymEngine::RCP<const SymEngine::Basic>>' requested here
    cereal::PortableBinaryOutputArchive{oss}(major, minor,
                                            ^
/usr/local/include/cereal/cereal.hpp:398:85: note: passing argument to parameter 'sharedPointer' here
      inline std::uint32_t registerSharedPointer(const std::shared_ptr<const void>& sharedPointer)
                                                                                    ^
1 error generated.
```

FreeBSD 13

**Comment by isuruf** [comment]
Which version of cereal is this?

**Comment by isuruf** [comment]
v1.3.1 made a backwards incompatible change, but did not change the version macro, so we can't easily distinguish between v1.3.0 and v1.3.1 to apply a fix. This was fixed in v1.3.2.
So, use v1.3.2 or v1.3.0, but don't use v1.3.1.

See https://github.com/USCiLab/cereal/issues/748
and https://github.com/symengine/symengine/blob/ad84b9677e814e6ac328d7127da092d4d8722afe/symengine/serialize-cereal.h#L283-L289

Duplicate of https://github.com/symengine/symengine/issues/1886

**Comment by yurivict** [comment]
cereal was version 1.3.1. It builds fine with cereal-1.3.2, thanks!

Please feel free to close this issue.
